Utils/orwell.js

```diff
@@ -33,5 +35,71 @@ module.exports = {
     return new Promise((resolve, reject) => {
       r.db('DFB').table('analytics').get(user).run().then(resolve).catch(reject)
     })
+  },
+  roleUsers: (guild, bot) => {
+    r.db("DFB").table("analytics").run().then((results) => {
+      results.forEach((row) => {
+        let totalDays = Object.keys(row.messages).length
+        let consecutiveDays = 3 // hardcoded until doug adds CD to schema
+        let member = guild.members.find(member => member.id === row.id)
+        if (!member) return;
+        
+        // is the user active?
+        let active = false
+
+        var roleWeights = [] // array of role weights for every role the user has
+
+        Object.entries(roles).forEach(([key, role]) => {
+          if (member.hasRole(key)) {
+            // if user has role, then get all dates in between role.decay and now
+            // if user has not interacted in those dates, then they are no longer active
+            let dates = Array.apply(null, new Array(role.decay)).map((v, i) => {
+              var d = new Date();
+              d.setDate(d.getDate() + i + 1 - 7)
+              d.setHours(0,0,0,0)
+              return d.getTime()
+            })
+            if (dates.some(date => date in row.messages)) active = true;
+          }
+
+          if (!active) {
+            if (member.hasRole(key)) roleWeights.push(role.rank)
+          } else if (totalDays && consecutiveDays >= role.threshold) {
+            if (member.hasRole(key)) return
+            if (role.message) {
+              member.openDM().then((channel) => {
+                channel.sendMessage(`Hey ${member.name}! ${role.message}`)
+              })
+            }
+            member.assignRole(key).then(() => { 
+              genlog.log(bot, bot.User, { 
+                message: `Added ${member.name}#${member.discriminator} to ${role.name}.`
+              })
+            })
+            return
+          }
+        })
+
+        if (!active) {
+          if (roleWeights === []) return; // has no roles :c

```

Comment:
you can't compare arrays like this in javascript lol

```js
> [] === []
< false
```

use `roleWeights.length`